tests/HeaderDataTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+use PHPUnit\Framework\TestCase;
+use Pantheon\EI\HeaderData;
+
+final class HeaderDataTest extends TestCase
+{
+  public function testConstructor(): void {
+    // Empty Constructor.
+    $headerData = new HeaderData();
+    $this->assertIsObject($headerData);
+
+    // Empty header data.
+    $headerData = new HeaderData([]);
+    $this->assertIsObject($headerData);
+
+    // Constructor with a small amount of array data.
+    $inputArray = [
+        'testConstructor' => 'constructor'
+    ];
+    $headerData = new HeaderData($inputArray);
+    $this->assertIsObject($headerData);
+  }
+
+  public function testGetHeader(): void {
+    $input = [
+      'HTTP_SHOULD_BE_FOUND' => 'Should be found',
+      'SERVER_NAME' => 'Entries without http_ should be ignored',
+      'HTTP_USER_AGENT' => 'Should become User_Agent'
+    ];
+    $headerData = new HeaderData($input);
+
+    $this->assertEquals($headerData->getHeader('Should-Be-Found'), 'Should be found');
+    $this->assertEmpty($headerData->getHeader('Server-Name'), 'SERVER_NAME should be ignored');
+    $this->assertEquals($headerData->getHeader('User-Agent'), 'Should become User_Agent');
+  }
+
+  public function testParseHeader(): void {
```

Comment:
This is an instance of where a descriptive docblock could be really helpful. Most of the assertions in this test are pretty straightforward but explaining _why_ we're testing what we're testing would be helpful for future devs looking at this file. This might also be a good place to use `@dataProvider`s.

Reply:
Great idea :+1: